src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final String COLOR = new ColorSupplier().getRandomColor();
+    private static final int LIMIT_NUMBER = 20;
+
+    public Figure getRandomFigure() {
+        switch (RANDOM.nextInt(4)) {

```

Comment:
The switch statement should have 5 cases for numbers 0 to 4. Since you are using `nextInt(4)`, it will only generate numbers from 0 to 3, missing one case. It should be `nextInt(5)` to cover all cases from 0 to 4 as per the checklist requirement.